src/dd/putbit.c

```diff
@@ -62,7 +62,8 @@ int putbit_write(struct putbit *pb, unsigned count, unsigned val)
 
 	for (unsigned i=0; i<count; i++) {
 
-		unsigned bit = (val >> (count-i-1)) & 0x1;
+		unsigned shift = count-i-1;
+		unsigned bit = (val >> shift) & 0x1;
```

Comment:
Looks like the shifting has to be limited (by mask size) to prevent shift count overflow `>=32 (sizeof(unsigned) * 8)`

Reply:
Or maybe a limit check for count is enough